pkg/gardenlet/controller/shoot/care/health.go

```diff
@@ -666,6 +680,30 @@ func (h *Health) CheckClusterNodes(
 	return nil, nil
 }
 
+func CheckNodeAgentLease(nodeList *corev1.NodeList, leaseList *coordinationv1.LeaseList, clock clock.Clock) error {
+
+	for _, node := range nodeList.Items {
+		var nodeLease *coordinationv1.Lease = nil
+		for _, lease := range leaseList.Items {
+			if leasehelper.ObjectName(node.Name) == lease.Name {
```

Comment:
We probably should not include the controller package of the node agent into the gardenlet. Let's better move this `ObjectName` function to `pkg/utils/gardener/machines.go` and call it `NodeAgentLeaseName`.